certs/files.go

```diff
@@ -14,94 +14,63 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package cloudinit
+package certs
 
-import (
-	"github.com/pkg/errors"
-	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha2"
-)
-
-// Certificates is a template struct to hold certificate data
-type Certificates struct {
-	CACert           string
-	CAKey            string
-	EtcdCACert       string
-	EtcdCAKey        string
-	FrontProxyCACert string
-	FrontProxyCAKey  string
-	SaCert           string
-	SaKey            string
-}
-
-func (cpi *Certificates) validate() error {
-	if !isKeyPairValid(cpi.CACert, cpi.CAKey) {
-		return errors.New("CA cert material in the ControlPlaneInput is missing cert/key")
-	}
-
-	if !isKeyPairValid(cpi.EtcdCACert, cpi.EtcdCAKey) {
-		return errors.New("ETCD CA cert material in the ControlPlaneInput is  missing cert/key")
-	}
+import "sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha2"
 
-	if !isKeyPairValid(cpi.FrontProxyCACert, cpi.FrontProxyCAKey) {
-		return errors.New("FrontProxy CA cert material in ControlPlaneInput is  missing cert/key")
-	}
-
-	if !isKeyPairValid(cpi.SaCert, cpi.SaKey) {
-		return errors.New("ServiceAccount cert material in ControlPlaneInput is  missing cert/key")
-	}
-
-	return nil
-}
+const (
+	rootOwnerValue = "root:root"
+)
 
-func certificatesToFiles(input Certificates) []v1alpha2.Files {
+func CertificatesToFiles(input Certificates) []v1alpha2.Files {
```

Comment:
Going to need to add godocs for every exported function

Reply:
you did this while i was reviewing, resolving

Reply:
we may have interpreted that differently. I don't think we should block this PR on that change. We can update to use multiple secrets in a follow on so we can integrate with cert-manager and give us some time to design the solution.

Reply:
If we can agree to block v1a2 on that new issue, then I am fine for this to merge. But if v1a2 gets released with secrets managed this way, then it creates upgrade and compatibility issues down the line which are completely avoidable.

Reply:
Ok, we'll work it out in the issue.

Reply:
https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/issues/133